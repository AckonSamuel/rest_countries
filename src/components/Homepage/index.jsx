import PropTypes from 'prop-types';
import Container from '@mui/material/Container';
import Stack from '@mui/material/Stack';
import { styled } from '@mui/material/styles';
import CountryList from '../Card';
import Searchbar from '../Searchbar';
import SelectFilter from '../SelectFilter';
import Navbar from '../Navbar';
import StyledContainer from '../StyledContainer';

const Homepage = ({ countries, textListener, filteredSearch, filterContinent, filterListener, searchText }) => {
    console.log(countries)
    return (
        <><Navbar />
        <StyledContainer>
            <Stack spacing={2} sx={{ marginTop: 2 }}>
                <Searchbar textListener={textListener} searchText={searchText} />
                <SelectFilter filterListener={filterListener} filterContinent={filterContinent} />
                <CountryList
                    countries={countries}
                    searchText={searchText}
                    filteredSearch={filteredSearch} />
            </Stack>
        </StyledContainer></>
    )
};

Homepage.propTypes = {
    countries: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)).isRequired,
    filteredSearch: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)).isRequired,
    searchText: PropTypes.string.isRequired,
    textListener: PropTypes.func.isRequired,
    filterListener: PropTypes.func.isRequired,
    filterContinent: PropTypes.string.isRequired,
};

export default Homepage;