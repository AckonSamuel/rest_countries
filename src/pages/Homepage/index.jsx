import PropTypes from 'prop-types';
import CountryList from '../../components/Card';
import Searchbar from '../../components/Searchbar';
import SelectFilter from '../../components/SelectFilter';
import Navbar from '../../components/Navbar';
import StyledContainer from '../../components/StyledContainer';

const Homepage = ({
  countries, textListener, filteredSearch, filterContinent, filterListener, searchText,
}) => (
  <>
    <Navbar />
    <StyledContainer>
      <Searchbar textListener={textListener} searchText={searchText} />
      <SelectFilter filterListener={filterListener} filterContinent={filterContinent} />
      <CountryList
        countries={countries}
        searchText={searchText}
        filteredSearch={filteredSearch}
      />
    </StyledContainer>
  </>
);

Homepage.defaultProps = {
  countries: [],
  filteredSearch: [],
  searchText: '',
  filterContinent: '',
  textListener: () => {},
  filterListener: () => {},
};

Homepage.propTypes = {
  countries: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  filteredSearch: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.any)),
  searchText: PropTypes.string,
  textListener: PropTypes.func,
  filterListener: PropTypes.func,
  filterContinent: PropTypes.string,
};

export default Homepage;
