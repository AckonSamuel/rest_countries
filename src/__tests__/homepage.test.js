import React from 'react';
import renderer from 'react-test-renderer';
import Homepage from '../pages/Homepage';

it('renders homepage component correctly', () => {
  const tree = renderer
    .create(
      <Homepage />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
