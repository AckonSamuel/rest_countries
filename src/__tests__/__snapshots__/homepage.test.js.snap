// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1p84f4n-MuiPaper-root"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-yhn5o0-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ip7gaz-MuiToolbar-root"
      >
        <h1
          className="MuiTypography-root MuiTypography-logo css-1c1gya-MuiTypography-root"
        >
          Where in the world?
        </h1>
        <div
          className="MuiBox-root css-0"
          onClick={[Function]}
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-returnHome MuiButton-returnHomePrimary MuiButton-sizeMedium MuiButton-returnHomeSizeMedium MuiButton-root MuiButton-returnHome MuiButton-returnHomePrimary MuiButton-sizeMedium MuiButton-returnHomeSizeMedium css-175qf1a-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              className="moon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M160 136c0-30.62 4.51-61.61 16-88C99.57 81.27 48 159.32 48 248c0 119.29 96.71 216 216 216 88.68 0 166.73-51.57 200-128-26.39 11.49-57.38 16-88 16-119.29 0-216-96.71-216-216z"
                fill="none"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
            <span
              className="MuiTypography-root MuiTypography-darkMode css-b8v4tk-MuiTypography-root"
            >
              Dark Mode
            </span>
          </button>
        </div>
      </div>
    </header>
  </div>,
  <div
    className="MuiBox-root css-45y7ag"
  >
    <form
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-pxp7br-MuiPaper-root"
    >
      <button
        aria-label="search"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </button>
      <div
        className="MuiInputBase-root MuiInputBase-colorPrimary css-84b1rr-MuiInputBase-root"
        onClick={[Function]}
      >
        <input
          aria-label="search country"
          className="MuiInputBase-input MuiInputBase-inputTypeSearch css-19fibzo-MuiInputBase-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search Country..."
          style={
            Object {
              "color": undefined,
            }
          }
          type="search"
        />
      </div>
    </form>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-bhskf2-MuiPaper-root"
    >
      <div
        className="MuiFormControl-root css-lvg1jp-MuiFormControl-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-kvi2wh-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          id="demo-simple-select-label"
        >
          Filter by Region
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1u6gncy-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          onClick={[Function]}
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="demo-simple-select-label demo-simple-select"
            className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="demo-simple-select"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Filter by Region
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <section
      id="country-list"
    />
  </div>,
]
`;
